src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1217,8 +1217,7 @@ private static Document ParseDocumentPartial(JsonOperationContext context, ref T
             if (fields.Contain(DocumentFields.LastModified))
                 result.LastModified = TableValueToDateTime((int)DocumentsTable.LastModified, ref tvr);
 
-            if (fields.Contain(DocumentFields.Flags))
-                result.Flags = TableValueToFlags((int)DocumentsTable.Flags, ref tvr);
+            result.Flags = TableValueToFlags((int)DocumentsTable.Flags, ref tvr);
```

Comment:
I think that the issue is that we did not request 'Flags' in one of the flows that specified flags (by default we are retrieving 'All').

Reply:
There is no scenarios where we _can_ avoid using `Flags`, we assert on it immediately afterward